pkg/compactor/bucket_compactor.go

```diff
@@ -650,11 +667,10 @@ func NewBucketCompactorMetrics(blocksMarkedForDeletion prometheus.Counter, reg p
 			Help: "Total number of failures when verifying min/max time ranges of compacted blocks.",
 		}),
 		blocksMarkedForDeletion: blocksMarkedForDeletion,
-		blocksMarkedForNoCompact: promauto.With(reg).NewCounter(prometheus.CounterOpts{
-			Name:        "cortex_compactor_blocks_marked_for_no_compaction_total",
-			Help:        "Total number of blocks that were marked for no-compaction.",
-			ConstLabels: prometheus.Labels{"reason": block.OutOfOrderChunksNoCompactReason},
-		}),
+		blocksMarkedForNoCompact: promauto.With(reg).NewCounterVec(prometheus.CounterOpts{
+			Name: "cortex_compactor_blocks_marked_for_no_compaction_total",
+			Help: "Total number of blocks that were marked for no-compaction.",
+		}, []string{"reason"}),
```

Comment:
We should initialize all series with used reasons to 0, to avoid gaps in our series. ("out of order" and "critical", other reasons seem to be unused.)